nsqworker/nsqwriter.py

```diff
@@ -81,11 +82,13 @@ def finish_pub(self, conn, data, topic, payload):
         # Parse conn and data to decide whether message failed or not
         if isinstance(data, Error) or conn is None or data != 'OK':
             # Message failed, re-send
+            MESSAGES_RETRY_COUNTER.labels(topic).inc()
             self.logger.error("Message failed, waiting {} seconds before trying again..".format(delay))
             # Take a short break and then try to resend the message
             if isinstance(payload, str):
                 self.io_loop.call_later(delay, self.send_message, topic, payload)
             elif isinstance(payload, list):
                 self.io_loop.call_later(delay, self.send_messages, topic, payload)
         else:
+            MESSAGES_SENT_COUNTER.labels(topic).inc()
```

Comment:
don't you want to count per (channel,topic)?

Reply:
messages are published to topics